src/github.com/matrix-org/dendrite/clientapi/storage/output_room_events_table.go

```diff
@@ -13,21 +15,28 @@ CREATE TABLE IF NOT EXISTS output_room_events (
     -- NB: 'serial' makes no guarantees to increment by 1 every time, only that it increments.
     --     This isn't a problem for us since we just want to order by this field.
     id BIGSERIAL PRIMARY KEY,
+    -- The event ID for the event
+    event_id TEXT NOT NULL,
     -- The 'room_id' key for the event.
     room_id TEXT NOT NULL,
     -- The JSON for the event. Stored as TEXT because this should be valid UTF-8.
     event_json TEXT NOT NULL,
-    -- A list of event IDs which represent a delta of added/removed room state.
-    add_state_ids TEXT[] NOT NULL,
-    remove_state_ids TEXT[] NOT NULL
+    -- A list of event IDs which represent a delta of added/removed room state. This can be NULL
+    -- if there is no delta.
+    add_state_ids TEXT[],
+    remove_state_ids TEXT[]
```

Comment:
It turns out giving it nil slices really makes it insert `NULL`. Since the Go idiom is to pass around `nil` slices, I've reverted the previous change which had these set to `NOT NULL` so we can actually insert rows into this table.